### PR TITLE
Log errors when versions can't be created

### PR DESCRIPTION
**Pull request by md5**
As discussed in #807, this PR updates `record_update` and `record_destroy` to log a warning listing all validation errors when a `Version` instance cannot be saved. It does not update `record_create` since that calls `Version.create!` instead of `Version.create` and should raise an error for any validation failures.

Due to Rubocop complaining, I have also slightly refactored `has_paper_trail.rb` to fix some of the complaints. There is still a complaint about the `InstanceMethods` module being too many lines, but I didn't see any easy way to reduce the line count of these changes without going too far off topic and into the weeds.

### CODE REVIEW

**Comment by jaredbeck** [comment]
> There is still a complaint about the InstanceMethods module being too many lines ..

Just bump up the number in rubocop_todo.yml .. it's a fight for another day :)

File: lib/paper_trail/has_paper_trail.rb [hunk 1]
```diff
@@ -397,12 +395,12 @@ def record_update(force = nil)
...
-            data[:transaction_id] = PaperTrail.transaction_id
-          end
+          add_transaction_id(data)
           version = send(self.class.versions_association_name).create merge_metadata(data)
-          update_transaction_id(version)
-          save_associations(version)
+          unless log_version_errors(version, :update)
+            update_transaction_id(version)
+            save_associations(version)
+          end
```
Comment (jaredbeck) [inline lib/paper_trail/has_paper_trail.rb]: I find this conditional hard to read. I'd prefer:

```ruby
if version.errors.any?
  log_version_errors(version, :update)
else
  update_transaction_id(version)
  # etc.
```

If rubocop complains about method complexity, just bump up the numbers; it's a fight for another day.

File: lib/paper_trail/has_paper_trail.rb [hunk 2]
```diff
@@ -638,6 +636,11 @@ def save_version?
         (if_condition.blank? || if_condition.call(self)) && !unless_condition.try(:call, self)
       end
 
+      def add_transaction_id(data)
+        return unless self.class.paper_trail_version_class.column_names.include?("transaction_id")
+        data[:transaction_id] = PaperTrail.transaction_id
+      end
```
Comment (jaredbeck) [inline lib/paper_trail/has_paper_trail.rb]: Nice method extraction. How about `add_transaction_id_to` as a method name? (note the "to"). Also, if it's easy to do this method extraction in a separate commit, that would be great. If not, don't worry about it.

File: lib/paper_trail/has_paper_trail.rb (hunk 1 above)
Comment (md5) [inline lib/paper_trail/has_paper_trail.rb]: That's what I had originally until RuboCop complained about `AbcSize`. I've changed it back, though subsequent edits seem to have removed the need to bump `AbcSize`.

File: lib/paper_trail/has_paper_trail.rb (hunk 2 above)
Comment (md5) [inline lib/paper_trail/has_paper_trail.rb]: Done in 71b93758bf191560b2fab0ea96b50a231c26007d

**Comment by md5** [comment]
I've refactored and bumped the `Metrics/ModuleLength` limit in `.rubycop_todo.yml` 👍

**Comment by jaredbeck** [comment]
Thanks Mike!

**Comment by md5** [comment]
:metal:
